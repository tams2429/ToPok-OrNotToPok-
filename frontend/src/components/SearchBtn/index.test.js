import { render, fireEvent } from '@testing-library/react'
import { SearchBtn } from './index'

describe('Search Button when btnType=`submit`(i.e. pokeSearch Button)', () => {
  const initialSubmitProps = {
    btnText: 'PokÃ© Search',
    btnType: 'submit',
    isDisabled: false,
  }

  it('should render properly and match snapshot when disabled', () => {
    const pokeSearchBtnComponent = render(<SearchBtn {...initialSubmitProps} isDisabled={true} />)
    expect(pokeSearchBtnComponent.container).toMatchSnapshot()
  })
  it('should render properly and match snapshot when not disabled', () => {
    const pokeSearchBtnComponent = render(<SearchBtn {...initialSubmitProps} />)
    expect(pokeSearchBtnComponent.container).toMatchSnapshot()
  })
})

describe('Search Button when btnType=`button`(i.e. Surprise Me Button)', () => {
  const mockHandleRandomSearch = jest.fn()

  const initialBtnProps = {
    btnText: 'Surprise Me',
    btnType: 'button',
    isDisabled: false,
    onClick: mockHandleRandomSearch,
  }
  it('should render properly and match snapshot', () => {
    const surpriseMeBtnComponent = render(<SearchBtn {...initialBtnProps} />)
    expect(surpriseMeBtnComponent.container).toMatchSnapshot()
  })

  it('should call the mockHandleRandomSearch function when the button is clicked', () => {
    const {getByText} = render(<SearchBtn {...initialBtnProps} />)
    fireEvent.click(getByText(initialBtnProps.btnText))
    expect(mockHandleRandomSearch).toHaveBeenCalledTimes(1)

  })
})

