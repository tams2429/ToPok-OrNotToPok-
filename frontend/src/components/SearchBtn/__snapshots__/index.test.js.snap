// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Button when btnType=\`button\`(i.e. Surprise Me Button) should render properly and match snapshot 1`] = `
<div>
  <button
    class="searchBtn "
    type="button"
  >
    Surprise Me
  </button>
</div>
`;

exports[`Search Button when btnType=\`submit\`(i.e. pokeSearch Button) should render properly and match snapshot when disabled 1`] = `
<div>
  <button
    class="searchBtn disabled"
    disabled=""
    type="submit"
  >
    Poké Search
  </button>
</div>
`;

exports[`Search Button when btnType=\`submit\`(i.e. pokeSearch Button) should render properly and match snapshot when not disabled 1`] = `
<div>
  <button
    class="searchBtn "
    type="submit"
  >
    Poké Search
  </button>
</div>
`;
